### PR TITLE
new functions in solana client

### PR DESCRIPTION
**Pull request by aalu1418**
* test case for checking tx deduplication
* add `getSignatureStatuses` and `simulateTransaction` to client

### CODE REVIEW

**Comment by aalu1418** [comment]
Note: there is a bug in decoding the response in solana-go for `simulateTransaction` so the current PR will not pass until solana-go is updated

Addressed here: https://github.com/gagliardetto/solana-go/pull/56

_RESOLVED_ 🎉

File: pkg/solana/client/client.go [hunk 1]
```diff
@@ -151,6 +153,43 @@ func (c *Client) GetFeeForMessage(msg string) (uint64, error) {
...
+	}
+	return res.Value, nil
+}
+
+// https://docs.solana.com/developing/clients/jsonrpc-api#simulatetransaction
+func (c *Client) SimulateTx(ctx context.Context, tx *solana.Transaction) (*rpc.SimulateTransactionResult, error) {
+	ctx, cancel := context.WithTimeout(ctx, c.contextDuration)
+	defer cancel()
+
+	res, err := c.rpc.SimulateTransactionWithOpts(ctx, tx, &rpc.SimulateTransactionOpts{
```
Comment (krebernisak) [inline pkg/solana/client/client.go]: Consider adding an optional input for `opts` so the user could use them, or if `nil` the fn uses these defaults.

File: pkg/solana/client/client_test.go [hunk 2]
```diff
@@ -112,3 +115,150 @@ func TestClient_Reader_ChainID(t *testing.T) {
...
+	// check signature statuses
+	time.Sleep(2 * time.Second) // wait for processing
+	statuses, err := c.SignatureStatuses(ctx, []solana.Signature{sigSuccess, sigFail})
+	assert.NoError(t, err)
+
+	assert.Nil(t, statuses[0].Err)
+	assert.NotNil(t, statuses[1].Err)
+}
+
+func TestClient_SendTxDuplicates_Integration(t *testing.T) {
```
Comment (krebernisak) [inline pkg/solana/client/client_test.go:203]: Great job confirming this!

**Review by krebernisak — DISMISSED** [review]
Looks good to go, once `solana-go` is updated. Nice catch there!

File: pkg/solana/client/client.go (hunk 1 above)
Comment (aalu1418) [reply pkg/solana/client/client.go]: added!

**Review by krebernisak — APPROVED** [review]

**Pull request merged**